docs/book/user-guide/tutorials/trigger-pipelines-from-external-systems.md

```diff
@@ -0,0 +1,637 @@
+    )
+    
+    print(f"Triggered pipeline run with ID: {run.id}")
+```
+
+#### Using the REST API:
+
+The REST API is ideal for external system integration, allowing you to trigger pipelines from non-Python environments:
+
+```bash
+# Step 1: Get the pipeline ID
+curl -X 'GET' \
+  'https://<YOUR_ZENML_SERVER>/api/v1/pipelines?name=training_pipeline' \
+  -H 'accept: application/json' \
+  -H 'Authorization: Bearer <YOUR_TOKEN>'
+
+# Step 2: Get the template ID using the pipeline_id
+curl -X 'GET' \
+  'https://<YOUR_ZENML_SERVER>/api/v1/run_templates?pipeline_id=<PIPELINE_ID>' \
+  -H 'accept: application/json' \
+  -H 'Authorization: Bearer <YOUR_TOKEN>'
+
+# Step 3: Trigger the pipeline with custom parameters
+curl -X 'POST' \
+  'https://<YOUR_ZENML_SERVER>/api/v1/run_templates/<TEMPLATE_ID>/runs' \
+  -H 'accept: application/json' \
+  -H 'Content-Type: application/json' \
+  -H 'Authorization: Bearer <YOUR_TOKEN>' \
+  -d '{
+    "steps": {
+      "load_data": {
+        "parameters": {
+          "data_url": "s3://production-bucket/latest-data.csv"
+        }
+      },
+      "train_model": {
+        "parameters": {
+          "model_type": "gradient_boosting"
+        }
+      }
+    }
+  }'
+```
+
+> Note: When using the REST API, you need to specify parameters at the step level, not at the pipeline level. This matches how parameters are configured in the Python client.
+
+### Security Considerations for API Tokens
+
+When using the REST API for external systems, proper token management is critical:
+
+```python
+from zenml.client import Client
+
+# Create a service account for automated triggers
+service_account = Client().create_service_account(
+    name="pipeline-trigger-service",
+    description="Service account for external pipeline triggering"
+)
+
+# Generate API token with appropriate permissions
+token = Client().create_service_account_token(
+    service_account.id,
+    name="production-trigger-token",
+    description="Token for production pipeline triggers"
+)
+
+print(f"Store this token securely: {token.token}")
+# Make sure to save this token value securely
+```
+
+Use this token in your API calls, and store it securely in your external system
+(e.g., as a GitHub Secret, AWS Secret, or environment variable). Read more
+about [service accounts and tokens](https://docs.zenml.io/api-reference/oss-api/getting-started#using-a-service-account-and-an-api-key).
+
+## Method 2: Building a Custom Trigger API
+
+If you're using the open-source version of ZenML or prefer a customized
+solution, you can create your own API wrapper around pipeline execution. This
+API can support both direct pipeline execution and run template triggering.
+
+The wrapper presented below is a simple example. Note that you need to have
+ZenML installed for it to work but also the requirements of the precise stack
+you're using (unless you're only triggering a run template, in which case
+`zenml` on its own is sufficient).
```

Comment:
Thats wrong, the run template has nothing to do with this at all....

Reply:
Why not? The example below shows how to trigger both a synchronous execution of
the pipeline as well as how to trigger a run template to execute (remotely). If
you're just triggering a remote template you'd never need to actually trigger
the pipeline locally so you don't need the dependencies. Or I guess I'm
misunderstanding something?